tests/test_cargo_publish.rs

```diff
@@ -84,7 +176,8 @@ test!(simple {
 });
 
 test!(git_deps {
-    let p = project("foo")
+    let root = paths::root().join("git_deps");
+    let p = repo(&root)
```

Comment:
Hm ok. This is a little worrisome because I know of some workflows where entire home directories are git repositories which means that Cargo projects not in a git repository may end up producing lots of spurious warnings like this. I guess `cargo new` creates new git repos though so maybe it's not too bad.

I think though it may be a case where it's easy-ish to handle by just checking to see if `Cargo.toml` is in version control at all. If you're publishing a crate and we discovered a git repo where the `Cargo.toml` is explicitly ignored, then you can probably ignore that git repo for status checks. I think that'll take care of these tests at least.